Essentials/src/net/ess3/api/events/MuteStatusChangeEvent.java

```diff
@@ -4,7 +4,35 @@
 
 
 public class MuteStatusChangeEvent extends StatusChangeEvent {
+    private Long timestamp;
+    private String reason;
+
+    @Deprecated
     public MuteStatusChangeEvent(IUser affected, IUser controller, boolean value) {
+        this(affected, controller, value, null, null);
+    }
+
+    public MuteStatusChangeEvent(IUser affected, IUser controller, boolean value, Long timestamp, String reason) {
         super(affected, controller, value);
+        this.timestamp = timestamp;
+        if (reason == null) {
+            this.reason = null;
+        } else {
+            this.reason = reason.isEmpty() ? null : reason;
```

Comment:
When will reason ever be 'empty'? Maybe if a stupid plugin makes it empty but then we should be witchhunting them, not adding this weird oneoff check here.

Reply:
It's a prevention measure to assure that the API is consistent; nothing wrong with this. Nonetheless, I changed this to a one-liner.